compiler/src/iree/compiler/ExternalInterfaces/EncodingExternalModels.cpp

```diff
@@ -0,0 +1,128 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/ExternalInterfaces/EncodingExternalModels.h"
+#include "iree/compiler/Dialect/Encoding/IR/EncodingDialect.h"
+#include "iree/compiler/Dialect/Encoding/IR/EncodingOps.h"
+#include "iree/compiler/Dialect/Encoding/IR/EncodingTypes.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/IR/Builders.h"
+
+namespace mlir::iree_compiler {
+namespace {
+
+struct ContractionAttrPropagationInterface
+    : public IREE::Encoding::EncodingPropagationAttrInterface::ExternalModel<
+          ContractionAttrPropagationInterface, IREE::Encoding::MatmulKAttr> {
+  bool isPropagable(Attribute attr, Value target) const {
+    auto encoding = dyn_cast<IREE::Encoding::MatmulKAttr>(attr);
+    if (!encoding) {
+      return false;
+    }
```

Comment:
It should always be castable by the [ExternalModel mechanism design](https://mlir.llvm.org/docs/Interfaces/#external-models-for-attribute-operation-and-type-interfaces) because we are adding the extension to the MatmulK attribute.

```suggestion
    auto encoding = cast<IREE::Encoding::MatmulKAttr>(attr);
```